packages/remix-express/server.ts

```diff
@@ -49,7 +49,8 @@ export function createRequestHandler({
     next: express.NextFunction
   ) => {
     try {
-      let request = createRemixRequest(req, res);
+      let { publicPath } = typeof build === "function" ? await build() : build;
+      let request = createRemixRequest(req, res, publicPath !== "/");
```

Comment:
@markdalgleish I'm less confident in this now since it's using `publicPath` which may or may not have come from `vite` - so it could potentially introduce a breaking change if existing Remix apps with a defined `publicPath` are using (and relying on) middlewares that mess with `req.url`.  We may need either a future flag, or ideally a way to detect that this is a vite build.  Maybe we could put a flag on `ServerBuild`?  Or is there something we can look for in there already that tells us with certainty that this was a vite build?

Reply:
We could add a `compiler` export to `ServerBuild` which is `'vite' | 'esbuild'`?